src/Transport/Sending/MessageSenderPool.cs

```diff
@@ -4,59 +4,75 @@
+            if (!sendersForDestination.TryDequeue(out var sender) || sender.IsClosed)
             {
                 // Send-Via case
+                // TODO: should have ServiceBusClientOptions { EnableCrossEntityTransactions = true }
                 if (receiverConnectionAndPath != (null, null))
                 {
-                    sender = new MessageSender(receiverConnectionAndPath.connection, destination, receiverConnectionAndPath.path, retryPolicy);
+                    var client = new ServiceBusClient(connectionString, tokenCredential, serviceBusClientOptions);
+
+                    sender = client.CreateSender(destination);
                 }
                 else
                 {
-                    if (tokenProvider == null)
+                    if (tokenCredential == null)
                     {
-                        sender = new MessageSender(connectionStringBuilder.GetNamespaceConnectionString(), destination, retryPolicy);
+                        //sender = new MessageSender(connectionStringBuilder.GetNamespaceConnectionString(), destination, retryPolicy);
+                        var client = new ServiceBusClient(connectionString, serviceBusClientOptions);
+
+                        sender = client.CreateSender(destination);
                     }
                     else
                     {
-                        sender = new MessageSender(connectionStringBuilder.Endpoint, destination, tokenProvider, connectionStringBuilder.TransportType, retryPolicy);
+                        var client = new ServiceBusClient(connectionString, tokenCredential, serviceBusClientOptions);
+
+                        sender = client.CreateSender(destination);
+                        //sender = new MessageSender(connectionStringBuilder.Endpoint, destination, tokenProvider, connectionStringBuilder.TransportType, retryPolicy);
                     }
                 }
             }
 
             return sender;
         }
 
-        public void ReturnMessageSender(MessageSender sender)
+        public void ReturnMessageSender(ServiceBusSender sender)
         {
-            if (sender.IsClosedOrClosing)
+            if (sender.IsClosed)
             {
                 return;
             }
 
-            var connectionToUse = sender.OwnsConnection ? null : sender.ServiceBusConnection;
-            var path = sender.OwnsConnection ? sender.Path : sender.TransferDestinationPath;
-
-            if (senders.TryGetValue((path, (connectionToUse, sender.ViaEntityPath)), out var sendersForDestination))
+            // var connectionToUse = sender.FullyQualifiedNamespace.OwnsConnection ? null : sender.ServiceBusConnection;
+            // var path = sender.OwnsConnection ? sender.Path : sender.TransferDestinationPath;
+            //
+            // if (senders.TryGetValue((path, (connectionToUse, sender.ViaEntityPath)), out var sendersForDestination))
```

Comment:
Send-via is done differently in Track 2.
To have cross entity transactional operation, `ServiceBusClientOptions` has to be instructed to use send-via.

Reply:
Got it, and seems to be done.